core/cas-server-core-web/src/main/java/org/apereo/cas/config/CasCoreWebConfiguration.java

```diff
@@ -30,6 +33,24 @@
     @Autowired
     private CasConfigurationProperties casProperties;
 
+    /**
+     * Get messages from one or more property files containing values
+     * that should be exposed to Thyme templates but don't need i18n.
+     * This bean is Autowired into {@link messageSource()}.
+     * @return PropertiesFactoryBean containing all common (non-i18n) messages
+     */
+    @Bean
+    public PropertiesFactoryBean casCommonMessages() {
```

Comment:
To clarify, what I find confusing is: Why do have a bean created as a factory, only to then mark as optional, only to then somehow have spring proxy the object from the factory, only to then set it on the resource bundle...when it can directly be set on the resource bundle? Seems a lot of needless maneuvering. 

If it makes things easier:

1. Nobody would ever need to customize this file, or add to it.
2. If they do, they come to the project and submit a PR.
3. Or they rip this all out and do it their own way. 

#3 is how it happens. Almost always.